Hyeon

kim-howard opened the pull request:
check!

kim-howard committed:
Add : Random Naming API

Add
Random Naming API Logic

get RandomNaming and Create Logic

kim-howard committed:
Merge branch 'master' into hyeon

kim-howard committed:
Remove : Some tests

kim-howard committed:
Add : test sample

but it dosen't work...

gksxodnd007 commented on src/test/java/org/seoul/kk/repository/TravelerNamingRepoTests.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Test
+    public void test(){
```
이테스트는 어떤 로직과 상관없이 무조건 통과하는 테스트 아닐까요?? Assertions의 static 메서드를 이용해서 랜덤 네이밍이 중복되지않는 값으로 나오는 테스트 로직이 들어가야 할것 같아요

gksxodnd007 commented on src/main/java/org/seoul/kk/dto/RegisterTravelerDto.java:
```diff
@@ -0,0 +1,15 @@
+public class RegisterTravelerDto {
+    private RandomNamingReturnDto namingInfo;
+    private String nickName;
+    private Object travelerProperty;
```
TravelerProperty enum클래스를 사용해주세요.

gksxodnd007 commented on src/main/java/org/seoul/kk/entity/TravelerNaming.java:
```diff
@@ -0,0 +1,27 @@
+    // and if number is 'two' property is noun
+    // 1 : 형용사 2 : 명사
+    @Column(name = "classification")
+    private Integer classification;
```
분류가 형용사인지 명사인지 나타내려고 하는건 enum값을 사용하면 어떨까요?? 이 컬럼에 1과 2가아닌 3이 저장되는 휴먼에러가 발생할 수 도있을 거같아요

gksxodnd007 commented on src/main/java/org/seoul/kk/entity/TravelerNaming.java:
```diff
@@ -0,0 +1,27 @@
+    private Integer classification;
+    // 들어가는 값
+    @Column(name = "property")
+    private String property;
```
이 컬럼은 어떤 의미인가요??

gksxodnd007 commented on src/main/java/org/seoul/kk/repository/TravelerNamingRepository.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.List;
+
+public interface TravelerNamingRepository extends JpaRepository<TravelerNaming,Long> {
+    List<TravelerNaming> findTravelerNamingsByClassification(Integer classification);
```
classification의 타입이 수정된다면 이것도 수정되어야 할 것같아요

gksxodnd007 commented on src/main/java/org/seoul/kk/service/TravelerServiceImpl.java:
```diff
@@ -1,18 +1,97 @@
+        for(int i=0;i<adjSize;i++) a_list.add(i);
+        Collections.shuffle(a_list);
+
+        List<TravelerNaming> nounProperties = travelerNamingRepository.findTravelerNamingsByClassification(2);
```
2가 어떤 값인지 주석을 보지않으면 판단하기 힘든 거같아요

gksxodnd007 commented on src/main/java/org/seoul/kk/service/TravelerServiceImpl.java:
```diff
@@ -1,18 +1,97 @@
+            TravelerNaming nowNamingObj= travelerNamingRepository.getOne(adjTraget);
+            String usedNounList = nowNamingObj.getUsedList();
+
+            for(int k=0;k<nounProperties.size();k++)
```
for문 조건에 nounProerties.size()함수를 호출하는 건 비효율적 인 것같아요 size를 변수에 저장해놓고 사용하는건 어떤가요?? 더 좋은 방법은 Stream과 람다를 사용하는 법도 있을 것같네요

gksxodnd007 commented on src/main/java/org/seoul/kk/service/TravelerServiceImpl.java:
```diff
@@ -1,18 +1,97 @@
+            travelerNamingRepository.save(adjective);
+
+        }else{
+            // throw Error
```
닉네임을 등록하는 과정에서 race condition을 막고싶다면 트랜잭션isolation level을 이용하는게 좋을 것같아요

gksxodnd007 reviewed (commented):
마스터를 이 브랜치에 rebase하거나 머지한 후에 다시 푸쉬해주세요

kim-howard committed:
Feat : Naming source C

Add Naming source Create Code

Test is Done

Should remove test Code

gksxodnd007 commented on src/main/java/org/seoul/kk/entity/TravelerNaming.java:
```diff
@@ -1,21 +1,22 @@
 
 @Data
 @Entity
+@Builder
```
@Builder애너테이션 이용시 @AllArgsContructor도 같이 사용해야 에러가 안나더라구요. @Builder에서 모든 변수를 가진 생성자를 이용해서 객체를 만들어요

kim-howard committed:
Merge branch 'master' into hyeon

# Conflicts:
#	src/main/java/org/seoul/kk/controller/TravelerController.java
#	src/main/java/org/seoul/kk/dto/RegisterTravelerDto.java
#	src/main/java/org/seoul/kk/service/TravelerService.java
#	src/main/java/org/seoul/kk/service/TravelerServiceImpl.java

kim-howard committed:
Add : Random source entity

kim-howard committed:
Merge branch 'master' into hyeon

kim-howard committed:
Add : Random source

kim-howard committed:
Add : Random source 2

kim-howard committed:
Add : Random Naming Service

Add Random Naming Return Service ( no test )

kim-howard committed:
Add & Refactoring : nicknameService ApiResponse

Refactoring service
Remove TempNickName

kim-howard committed:
Remove : Test Code

kim-howard committed:
Feat & Refactor : Register Traveler Naming

if User register Traveler then register that source to Naming Used List

and Refactoring DTO

gksxodnd007 committed:
merge master

gksxodnd007 committed:
generate random nickname api
